app/questionnaire/questionnaire_store_updater.py

```diff
@@ -164,16 +167,21 @@ def remove_list_item_and_answers(self, list_name: str, list_item_id: str) -> Non
 
         self._progress_store.remove_progress_for_list_item_id(list_item_id=list_item_id)
 
+        if remove_block_id := self._schema.get_remove_block_id_for_list(list_name):
```

Comment:
Maybe better/clearer to handle the dependency capture in a separate method? As now this has been updated the title of this method is misleading in that it is no longer just removing from the progress store?

Reply:
I like this idea 👍 it splits very cleanly into the removal and dependencies so made the new method as suggested